parsing/location.mli

```diff
@@ -173,17 +181,17 @@ val best_toplevel_printer: unit -> report_printer
 
 (** {2 Printing a [report]} *)
 
-val print_report: formatter -> report -> unit
+val print_report: Misc.Log.t -> report -> unit
 (** Display an error or warning report. *)
 
-val report_printer: (unit -> report_printer) ref
```

Comment:
Before this PR the API allows user to change the way reports are printed. There are two choices for this evolution:
- we allow users to redefine both the "terminal" printer and the "json" printer, at the cost of a change of type in the API. This is what this PR currently done.
- we only allow users to redefine the "terminal" printer, and the API stays the same.

I find the second choice slightly more tempting, given that there is little justification for changing the way the json values is printed (it is easy to read it back and transform after the fact if necessary).

Reply:
I went with the full backward compatibility option, without hooks the json printer for now. We can refine the API later if this is needed.